etc/scripts/timetable/export.js

```diff
@@ -0,0 +1,152 @@
+
+    .demand('o')
+    .alias('o', 'output')
+    .describe('o', 'The JSON file to write the tree too')
+
+    .demand('a')
+    .alias('a', 'app')
+    .describe('a', 'The id of the application for which the tree should be exported')
+
+    .argv;
+
+// Initialize the app server
+GrassHopper.init(config, function(err) {
+    if (err) {
+        log().error({'err': err}, 'Failed to spin up the application container');
+        process.exit(1);
+    }
+
+    log().info('Getting all the courses');
+    getCourses(function(courses) {
+
+        log().info('Exporting all the courses');
+        exportCourses(courses, function(courses) {
+
+            log().info('Flushing courses to disk');
+            writeCourses(courses, function() {
+                process.exit(0);
+            });
+        });
+    });
+});
+
+/**
+ * Get all the courses in the application
+ */
+var getCourses = function(callback) {
+    OrgUnitDAO.getOrgUnits(argv.app, null, ['course'], null, function(err, orgUnits) {
+        if (err) {
+            log().error({'err': err}, 'Failed to get the root nodes');
+            process.exit(1);
+        }
+
+        return callback(orgUnits);
+    });
+};
+
+/**
+ * Export the organisational units, event series and events under the courses
+ */
+var exportCourses = function(courses, callback, _exportedCourses) {
+    _exportedCourses =  _exportedCourses || [];
+    if (_.isEmpty(courses)) {
+        return callback(_exportedCourses);
+    }
+
+    var course = courses.pop();
+    log().info({'id': course.id}, 'Exporting %s', course.displayName);
+    OrgUnitDAO.exportOrgUnit(course, 'json', function(err, exportedCourse) {
+        if (err) {
+            log().error({'err': err, 'course': course.id}, 'Failed to export a course');
+            process.exit(1);
+        }
+
+        // Remove the id properties from the exported courses
+        visitOrgUnit(exportedCourse);
+
+        _exportedCourses.push(exportedCourse);
+        exportCourses(courses, callback, _exportedCourses);
+    });
+};
+
+/**
+ * Remove the `id` and `ParentId` from every exported organisational unit
+ */
+var visitOrgUnit = function(orgUnit) {
+    orgUnit.metadata = orgUnit.metadata || {};
+    orgUnit.metadata.exportedId = orgUnit.id;
+    delete orgUnit.id;
+    delete orgUnit.ParentId;
+
+    _.each(orgUnit.children, visitOrgUnit);
+    _.each(orgUnit.series, function(series) {
+        series.metadata = series.metadata || {};
+        series.metadata.exportedId = series.id;
+        delete series.id;
+        _.each(series.events, function(event) {
+            delete event.id;
+        });
+    });
+
+    // Splice off borrowed series so they don't get auto-created when importing
+    orgUnit.borrowedSeries = _.filter(orgUnit.series, function(series) {
+        return (series.borrowedFrom);
+    });
+    orgUnit.series = _.filter(orgUnit.series, function(series) {
+        return (!series.borrowedFrom);
+    });
```

Comment:
I think `_.partition` was intended for separating a list into 2 divisions, so we could use that instead